src/components/content/document/patient-documents.tsx

```diff
@@ -1,26 +1,36 @@
 import cx from "classnames";
-import { useRef } from "react";
+import { useMemo, useRef } from "react";
 import { patientDocumentColumns } from "./helpers/columns";
 import { getDateRangeView } from "../resource/helpers/view-date-range";
 import { useResourceDetailsDrawer } from "../resource/resource-details-drawer";
 import { ResourceTable } from "../resource/resource-table";
 import { ResourceTableActions } from "../resource/resource-table-actions";
 import { withErrorBoundary } from "@/components/core/error-boundary";
 import { useCTW } from "@/components/core/providers/use-ctw";
-import { usePatientDocument } from "@/fhir/document";
+import { RowActionsProps } from "@/components/core/table/table";
+import { getBinaryDocument } from "@/fhir/binaries";
+import { usePatientDocuments } from "@/fhir/document";
 import { DocumentModel } from "@/fhir/models/document";
 import { useFilteredSortedData } from "@/hooks/use-filtered-sorted-data";
+import { useBaseTranslations } from "@/i18n";
 
 export type PatientDocumentProps = {
   className?: string;
   enableFQS?: boolean;
+  // could call this additionalRowActions and make it an array
+  onAddToRecord?: (document: DocumentModel, binary: fhir4.Binary) => void;
```

Comment:
maybe with something like
```suggestion
  rowAction: RowAction;
```
or 
```suggestion
  rowActions: RowAction[];
```
where `RowAction` is
```tsx
type RowAction = {
  text: string;
  onAction: (document: DocumentModel, binary: fhir4.Binary) => void;
  className?: cx.Argument;
};
```
and that way consumer of the library who is implementing this action is required to add
 - `text` - Text that appears on the button
 - `onAction` - Callback to handle the click, may be async, if this errors, we should show this error message in the UX.
 - `className` - optionally add a classname to the button